### DISCUSSION TITLE
Suggestion: Integrate AI chatbots into Side Panels

### DISCUSSION THREAD

**Discussion by kapuchai** [zen-browser/desktop#1691] [Ideas]
Latest Firefox (130) introduced Firefox Labs, which has a neat AI chatbots feature, that is supposed to open in "sidebar".
At this moment in ZEN, the side panel with chosen chatbot is accessible only through Firefox Labs settings page, where the user turns the feature on/off.
So I thought, is this something that could be integrated into ZEN's Side Panels? Or maybe it should have at least an independent icon in sidebar, so it can be accessed at all?

**Comment by radu-corbu** [comment]
I just installed Zen Browser and discovered the new Firefox Labs feature. I was so confused when the AI chat setting mentioned it would be added the sidebar, but no such option was actually added in Zen's sidebar.

I thought maybe there's another sidebar I'm unaware of. I also tried looking into Zen Browser's keyboard shortcuts to see if maybe I could press a key combination to bring up the AI chat. No such luck.

Please consider integrating the AI chat into Zen's sidebar. I find myself asking Claude/ChatGPT questions on a regular basis and it would make using them much easier if a chat panel was integrated into the browser.

**Comment by polsvoice** [comment]
I would like to see it integrated into a separate sidebar similar to Brave's, which has tabs for an AI, Bookmarks, and a reading list.

**Comment by supunTE** [comment]
You can enable AI chatbot feature from firefox in zen by setting `browser.ml.chat.enabled` to `true` in `about:config`

**Reply by atleeit** [reply]
Thanks mate, all working, restart browser and CTRL+ALT+X opens the sidebar required.

**Comment by rahatzamancse** [comment]
The `browser.ml.chat.enabled` is just a small tab/window for the websites. It has no integration with the tabs I have open, the tab I am on, or the website content. Brave's AI integration takes the entire context into account and is very convenient. I would like a similar feature to that.

**Reply by atleeit** [reply]
First off keep in mind Brave and Zen are two completely different browsers built using different backends, Are you referring to Braves Leo tab focus mode?

**Reply by rahatzamancse** [reply]
Leo is braves default AI model,  but it can also use other models like gpt or claude. And all those models work the same way Leo does, like having website access and other tabs contexts etc. 

I understand that the implementation is not easy since they are very different, brave is chromium-based and zen is firefox-based, so backend is completely different.  I'm just bumping this feature request.

**Comment by jmuellers** [comment]
Christ, I came here thinking this was the last browser that will leave me alone with AI shit and just let me browse. That's why I removed Chrome and Firefox once and for all. Did I get wrong what Zen browser is all about?

**Reply by GazzD** [reply]
I mean... if it is optiona,l what is the problem? Firefox have [AI Controls](https://blog.mozilla.org/firefox/ai-controls/) to hide from you anything related to AI 👍

**Reply by rahatzamancse** [reply]
> Christ, I came here thinking this was the last browser that will leave me alone with AI shit and just let me browse. That's why I removed Chrome and Firefox once and for all. Did I get wrong what Zen browser is all about?

Hating AI in 2026 is like being mad when smartphones got GPS. Sure, some apps abused it and tracked people creepily, but the technology itself became basic infrastructure. Within 2-3 years, everything you see/use will be integrated with AI, or get doomed for not using AI.

**Reply by jmuellers** [reply]
You are both right, and I apologize for my language.

My motivation to try out Zen browser stems from being overwhelmed with new technology (especially the fast paced integration of AI into everything) - I don't mind admitting that.
From what I read about Zen browser (or at least, what the name promised to me) I expected a software that can cater to "people like me" who just want to browser the web, and who don't have the mental capacity to deal with AI integration and the like. New buttons, sidebars, options I have to take care of turning off, that's really giving me a hard time when I just want to browse.

So, in the end, it is just a matter of having had the wrong expectation. I still hope you can sympathize to some extend.

**Reply by kapuchai** [reply]
Since Zen is based on Firefox, you can easily opt-out from any future AI features if you'd like (about:preferences#ai)

**Reply by jmuellers** [reply]
Yes, I understand that now. I'm coming from Firefox, where I went through those steps - and it made me so angry that I looked for another browser. As I said - it's just a case of bad expectation management.

I thought Zen is a browser with minimal feature set, so that I can be "zen" and don't have to take care of any kind of new "features" (like AI, like Firefox suddenly advertising some VPN, ...). 
Instead my understanding is now that it is mainly Firefox with a nice added layer of design - which is perfectly fine, just not what I expected :-)

**Reply by MaximilianZenz** [reply]
Yeah i hope too that ai will not be integrated in the browser. I don't see any reason for it. For everybody that wants it just pin the ai provider of your choice as an essential tab. But no need to integrate it into the browser itself. Especially since there are so many important features still missing from zen. This would be a complete waste of dev time at the moment.

**Reply by atleeit** [reply]
There are plenty of people that would like to have it, If AI is a part of the Firefox base, the devs won't have to do anything, it will come included. 

AI, regardless if people don't want to adopt it, this is the way of the future, yes it is , but if you don't, that is ok, you don't have to use products that have it, but don't take it away from people that do find it useful. Most computers are powerful enough to handle AI tasks without slowing it down, it is 2026 not 2010 lol.

**Reply by atleeit** [reply]
> > Hating AI in 2026 is like being mad when smartphones got GPS. Sure, some apps abused it and tracked people creepily, but the technology itself became basic infrastructure. Within 2-3 years, everything you see/use will be integrated with AI, or get doomed for not using AI.

You understand completely. People used to build cars in factories, now robots do it all. I'm sure workers went on strike back then but it didn't change the way of the future, the same goes for AI. Billions of dollars building data ceres all over the world, they aren't going to bail on making it work, and integrating it into everything.

**Comment by Craftgenix1** [comment]
As I've stated in a previous discussion, and as is shown in this comment section, I believe that adding ai integration into the browser would only be detrimental to the browser as a whole. not only would it be extra bloatware for some users, but it could turn users away from using the browser at all, even people who have been using the browser for a long time. I don't believe that the demand is truly high enough for this to be an efficient cost/benefit relationship, as I think that the majority of people asking for ai features are a very, _very_ vocal minority, and they shouldn't be the main basis of company opinion. Furthermore, when thinking about adding risky features such as this, I find it prudent that you consider putting out a survey to find your userbase's true opinions before doing something that could permanently scar your reputation in the eyes of some.

**Reply by Craftgenix1** [reply]
Oh wait, this is 2 years old. whoops

**Reply by rahatzamancse** [reply]
> I don't believe that the demand is truly high enough for this to be an efficient cost/benefit relationship, as I think that the majority of people asking for ai features are a very, _very_ vocal minority, and they shouldn't be the main basis of company opinion. Furthermore, when thinking about adding risky features such as this, I find it prudent that you consider putting out a survey to find your userbase's true opinions before doing something that could permanently scar your reputation in the eyes of some.

@Craftgenix1 Do you have any backing on saying that people asking for AI features is the minority (and vocal?)? Was there any survey done? Maybe you can give a source? Because i have source that says the complete opposite of what you are saying.

**Reply by Craftgenix1** [reply]
if you had a source then why don't you put it here?

**Reply by rahatzamancse** [reply]
@Craftgenix1 Look, you made a claim, and I just asked you to back it up. There is no need to be defensive and throw my own words back at me. We both want Zen Browser to be successful, and I think AI features in 2026 and beyond are inevitable. Many products and companies fail to adapt to emerging technologies and get left behind (see Nokia, Kodak, etc.). I don't want Zen to be one of them. You can agree to disagree, but you and I are not the only users who will be affected by the decision about AI features.

Here you go on my sources:

https://www.capgemini.com/news/press-releases/71-of-consumers-want-generative-ai-integrated-into-their-shopping-experiences/
https://business.adobe.com/blog/generative-ai-powered-shopping-rises-with-traffic-to-retail-sites
https://www.deloitte.com/us/en/insights/industry/telecommunications/connectivity-mobile-trends-survey.html
https://yougov.com/en-us/articles/50614-ai-on-smartphones-what-features-do-consumers-value-most
https://www.salesforce.com/news/stories/ai-customer-research/


I bet you can also show "some" reports, like the ones above, that say otherwise. But here is a much stronger and more trustworthy source that you cannot find online against AI. Stack Overflow conducted a survey and found that most people are AI-friendly and would switch to products with AI features, leaving products that don't have them eventually extinct. 

Source: https://survey.stackoverflow.co/2025/ai/

<img width="1039" height="740" alt="Screenshot 2026-06-26 at 7 04 10 AM" src="https://github.com/user-attachments/assets/6149f8fd-d205-484b-b8d3-0470f350f433" />

Also see my other comment: [#discussioncomment-16859430](https://github.com/zen-browser/desktop/discussions/1691#discussioncomment-16859430)

**Reply by Craftgenix1** [reply]
All of those are .com domains, meaning none of them are reputable sources because they're all trying to sell you something. I also looked into your graphs, and without the date it's hard to tell, but some of these could be inflated because of employers forcing their employees to use ai, as well as ai still being a bit of a new, and 'novel' concept (though by now it's mostly worn off I must admit). I was simply pointing out how just saying you have sources and then not linking them is a bit iffy. my source is purely just my experience, with how most people who really know what they're talking about when they talk about ai really really hate it, as well as how comically evil datacenters are being (taking water away from towns that need it [it evaporates and lands somewhere else], and the massive power draw pushing up the cost of living). my main point of 'ai bros' being a vocal minority pretty much boils down to a few main points:
1. I hear about y'all a LOT more than I see you
2. the thing that your archetype is so obsessed with spreads a lot of misinformation
3. It's harmful to most creative fields
4. it's harmful to the areas where it's stored
5. I still have a bit of faith in humanity, crazy as that is, and I don't think the majority of people just overlook all of that and keep using that barely-working product.
